Checks/PWR003/solution.f90

```diff
@@ -0,0 +1,25 @@
+! PWR003: Explicitly declare pure functions
+
+! Returns the gravity on Earth in m/s2
+pure function gravity_pure()
+  use iso_c_binding, only : c_double
+  implicit none
+  real(kind=c_double) :: gravity_pure
+  gravity_pure = 9.8
+end function gravity_pure
+
+! Computes the weight of each object in a vector
+subroutine solution_f(n, M, W) bind(c)
+  use iso_c_binding, only : c_int, c_double
+
+  implicit none
+  integer(kind=c_int) :: i
+  integer(kind=c_int), intent(in), value :: n
+  real(kind=c_double) :: gravity_pure
+  real(kind=c_double), dimension(1:n), intent(in) :: M
```

Comment:
This `gravity_pure` variable would not be necessary either.

Reply:
Idem